File: lib-extension/src/main/java/com/xiaoyv/flexiflix/extension/java/model/FlexMediaUser.kt
package com.xiaoyv.flexiflix.extension.java.model

import android.os.Parcelable
import com.xiaoyv.flexiflix.extension.java.utils.UNKNOWN_STRING
import kotlinx.parcelize.Parcelize

/**
 * [FlexMediaUser]
 *
 * @author why
 * @since 5/8/24
 */
@Parcelize
data class FlexMediaUser(
    var id: String = UNKNOWN_STRING,
    var name: String = UNKNOWN_STRING,
    var avatar: String = UNKNOWN_STRING,
    var role: String = UNKNOWN_STRING
) : Parcelable

File: web-nodejs/extension-common/src/model/FlexMediaUser.ts
/**
 * 发布条目的用户
 *
 * @author why
 * @since 5/8/24
 */
export interface FlexMediaUser {
    /**
     * 用户ID
     */
    id: string

    /**
     * 用户名称
     */
    name: string

    /**
     * 用户头像
     */
    avatar: string

    /**
     * 角色
     */
    role: string | undefined
}
